Tensile/TensileCreateLibrary.py

```diff
@@ -682,6 +684,64 @@ def noBuildError(kernel):
     return list(filter(noBuildError, kernels))
 
 
+def getKernelSourceAndHeaderCode(ko: KernelWriterBase) -> Tuple[int, List[str], List[str], str]:
+    """Get the source and header content for a kernel object.
+
+    Arguments:
+        ko: Kernel object to extract content from.
+
+    Returns:
+        Tuple of data: (error code, source code, header code, kernel name)
+
+    Notes:
+        - This function is very similar to **processKernelSource**---the functionality
+        should be merged.
+    """
+    name = ko.getKernelName()
+    err, src = ko.getSourceFileString()
+    hdr = ko.getHeaderFileString()
+    return err, [CHeader, src], [CHeader, hdr], name
```

Comment:
https://pypi.org/project/result/

maybe in the future...

Reply:
I would love to add then instead of using magic return codes. However, it adds another dependency which, as we've learned, is a heavier lift that expected.